src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,243 +32,218 @@
-    Download fewsnet data that covers `region_name`.
 
-    Parameters
-    ----------
-    date_pub : date
-        date for which the data should be downloaded
-        only the year and month part are used
-        this commonly refers to the month of the Current Situation period
-    region_name : str
-        name of the region to which the `iso2` belongs,
-        e.g. "east-africa"
-    region_code : str
-        code that refers to the `region_name,
-        e.g. "EA"
-    output_dir : Path
-        path to dir to which the data should be written
-    use_cache : bool
-        if True, don't download if output_dir already exists
-
-    Returns
-    -------
-    region_data : str
-        If region data exists, return the saved dir else return None
-    """
-    url_region_date = _BASE_URL_REGION.format(
-        region_code=region_code.upper(),
-        region_name=region_name,
-        YYYY=date_pub.year,
-        MM=date_pub.month,
-    )
-
-    output_dir_region = (
-        output_dir / f"{region_code.upper()}{date_pub.strftime('%Y%m')}"
-    )
-    zip_filename_region = f"{output_dir_region.name}.zip"
-    if not output_dir_region.exists() or use_cache is False:
-        region_data = _download_zip(
-            url=url_region_date,
-            zip_filename=zip_filename_region,
-            output_dir=output_dir_region,
+        # FN data can be regional, and thus we save the raw data
+        # in "glb" instead of in the iso3 folder, even though part of it is at
+        # country level
+        self._raw_base_dir = self._get_base_dir(
+            is_public=True,
+            is_raw=True,
+            is_global=True,
         )
-    else:
-        region_data = output_dir_region
-    return region_data
 
+        self._iso2 = Country.get_iso2_from_iso3(self._country_config.iso3)
+        if self._iso2 is None:
+            logger.error("No ISO2 found for the given ISO3. Check your ISO3.")
+
+    def download(
+        self,
+        date_pub: Union[date, str],
+        clobber: bool = False,
+    ) -> Path:
+        """
+        Retrieve the raw fewsnet data.
+
+        Depending on the region and date, this data is published per region or
+        per country. This function retrieves the country data
+        if it exists, and else the regional data for `date_pub`.
+
+        Parameters
+        ----------
+        date_pub : date or str
+            date for which the data should be downloaded
+            only the year and month part are used
+            this commonly refers to the month of the Current Situation period
+            if str, it should be in ISO 8601 format, e.g. '2021-10-01'
+        clobber : bool, default = False
+            If True, overwrites existing raw files
+
+        Examples
+        --------
+        >>> download(date_pub="2020-10-01")
+        """
+        # convert to datetime if str
+        if not isinstance(date_pub, date):
+            date_pub = datetime.strptime(date_pub, _DATE_FORMAT)
```

Comment:
Yeah good point. I am rethinking and do you think it would make more sense to split `date_pub` into two input args `pub_year` and `pub_month`? As we never use the day of the month, so this might be confusing to the user?

Reply:
Oo yes I like that idea a lot!